ca_driver/src/create_driver.cpp

```diff
@@ -1,28 +1,34 @@
 #include <tf/transform_datatypes.h>
 #include "create_driver/create_driver.h"
 
-CreateDriver::CreateDriver(ros::NodeHandle& nh) : nh_(nh), priv_nh_("~")
+CreateDriver::CreateDriver(ros::NodeHandle& nh) : nh_(nh), priv_nh_("~"), model_(create::RobotModel::CREATE_2)
 {
   bool create_one;
+  std::string robot_model_name;
   priv_nh_.param<double>("loop_hz", loop_hz_, 10.0);
   priv_nh_.param<std::string>("dev", dev_, "/dev/ttyUSB0");
   priv_nh_.param<bool>("create_1", create_one, false);
+  priv_nh_.param<std::string>("robot_model", robot_model_name, "CREATE_2");
   priv_nh_.param<double>("latch_cmd_duration", latch_duration_, 0.2);
   priv_nh_.param<bool>("publish_tf", publish_tf_, true);
 
   if (create_one)
   {
-    model_ = create::CREATE_1;
-    priv_nh_.param<int>("baud", baud_, 57600);
-    ROS_INFO("[CREATE] Create 1 model selected");
+    robot_model_name = "CREATE_1";
   }
-  else
+
+  try
   {
-    model_ = create::CREATE_2;
-    priv_nh_.param<int>("baud", baud_, 115200);
-    ROS_INFO("[CREATE] Create 2 model selected");
+    model_ = robot_model_name;
+    ROS_INFO_STREAM("[CREATE] \"" << robot_model_name << "\" selected");
+  } catch (std::exception& e)
```

Comment:
Instead of catching an exception, let's just check if the string matches one of the three possible robot types and initialize the model accordingly